storage/src/mmr/mod.rs

```diff
@@ -67,3 +67,11 @@ mod hasher;
 mod iterator;
 pub mod mem;
 pub mod verification;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid element position")]
+    InvalidElementPosition,
```

Comment:
nit: might be worth adding args and renaming this to `ElementPruned(u64)`

You can change the pragma to be `#[error("requested element is before oldest known: {0}")]`

Reply:
done